### PR TITLE
Implements Merciless

### PR DESCRIPTION
**Pull request by NxKarim**
- Implements `ConditionalCritAbAttr `, which given the pokemon attack condition guarantees a critical hit.
- Merciless checks for Poisoned or Intoxicated targets.
- Fixes BlockCritAbAttr not preventing guaranteed critical hits.

### CODE REVIEW

File: src/data/ability.ts [hunk 1]
```diff
@@ -1659,6 +1659,15 @@ export class MultCritAbAttr extends AbAttr {
 }
 
 
+export class CritIfTargetPoisonedAbAttr extends AbAttr {
+  apply(pokemon: Pokemon, passive: boolean, cancelled: Utils.BooleanHolder, args: any[]): boolean {
+    const target = (args[1] as Pokemon);
+    if(target.status.effect === StatusEffect.TOXIC || target.status.effect === StatusEffect.POISON)
+    (args[0] as Utils.BooleanHolder).value = true;
+    return true;
+  }
```
Comment (Xavion3) [inline src/data/ability.ts:1668]: Apply functions should return true if they did something and false otherwise. Also shouldn't line 1666 here be indented?

**Review by Xavion3 — CHANGES_REQUESTED** [review]

File: src/data/ability.ts (hunk 1 above)
Comment (NxKarim) [reply src/data/ability.ts]: yeah, done

File: src/data/ability.ts [hunk 2]
```diff
@@ -1823,6 +1823,17 @@ export class MultCritAbAttr extends AbAttr {
 }
 
 
+export class CritIfTargetPoisonedAbAttr extends AbAttr {
```
Comment (TempsRay) [inline src/data/ability.ts:1826]: Going forward, can you add doc on new classes?

**Review by TempsRay — COMMENTED** [review]
Code looks okay to me. Do you have any screenshots or video of testing this in action? And how it interacts with moves/abilities that blocks crits?

File: src/data/ability.ts (hunk 2 above)
Comment (NxKarim) [reply src/data/ability.ts:1829]: Yes, code comments added.

**Comment by NxKarim** [comment]
https://github.com/pagefaultgames/pokerogue/assets/43686802/b2e1446e-7dd1-4750-a5f4-9395ea5bc3ae

**Review by bennybroseph — CHANGES_REQUESTED** [review]
I think this should be a Crit if Status and pass in a status on the constructor. Could we modify this?

**Comment by shayebeadling** [comment]
Going off of the point made above, you could even do a conditional crit attr, similar to how we use our MovePowerBoostAttr. It would be super clean, and allow any "crit if x" stuff to work the same way.
image

**Comment by bennybroseph** [comment]
Nice, I do think that's the best solution

**Review by bennybroseph — APPROVED** [review]

**Pull request merged**